### PR TITLE
8339148: Add os::active_physical_processor_count() API

### PR DESCRIPTION
**Pull request by jerboaa**
Please review this rather trivial patch for adding `os::active_physical_processor_count()` function. It is useful to have that info on Linux for the container code which bounds the CPU limit above by that value. Also, the old `os::Linux::active_processor_count()` was protected and had a bit of a misleading name. As a bonus, we can get rid of some `friend class` smells.

**Testing**:
- [x] GHA
- [x] Container tests and cgroup gtests on linux x86_64

Thoughts?

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Integration blocker
&nbsp;⚠️ Title mismatch between PR and JBS for issue [JDK-8339148](https://bugs.openjdk.org/browse/JDK-8339148)

### Issue
 * [JDK-8339148](https://bugs.openjdk.org/browse/JDK-8339148): Make os::Linux::active_processor_count() public (**Enhancement** - P3) ⚠️ Title mismatch between PR and JBS.

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20768/head:pull/20768` \
`$ git checkout pull/20768`

Update a local copy of the PR: \
`$ git checkout pull/20768` \
`$ git pull https://git.openjdk.org/jdk.git pull/20768/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20768`

View PR using the GUI difftool: \
`$ git pr show -t 20768`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20768.diff">https://git.openjdk.org/jdk/pull/20768.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20768#issuecomment-2318315683)

### CODE REVIEW

File: src/hotspot/os/linux/os_linux.cpp [hunk 1]
```diff
@@ -4829,7 +4829,9 @@ static int get_active_processor_count() {
   return cpu_count;
 }
 
-int os::Linux::active_processor_count() {
+// Return the active CPUs irrespective of the system being containerized
+// or not. Thus, this returns the host active cpus in a container.
```
Comment (dholmes-ora) [inline src/hotspot/os/linux/os_linux.cpp:4833]: No it doesn't! The comment for the function you call states clearly:
```
// Get the current number of available processors for this process.
```
This is not related to "physical count" at all!

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
I'm not clear exactly what you need here but "physical" is not what you are looking up.

File: src/hotspot/os/linux/os_linux.cpp (hunk 1 above)
Comment (jerboaa) [reply src/hotspot/os/linux/os_linux.cpp:4833]: OK. Happy to change the name to something more suitable. The idea is to have a function that returns the "available" cpus of the node/host whether in container or not. How about `available_host_processor_count()`?

Comment (jerboaa) [reply src/hotspot/os/linux/os_linux.cpp:4833]: Just to be clear. It's not a goal to introduce new semantics.  What we need is an public accessible way to query this information.

Comment (dholmes-ora) [reply src/hotspot/os/linux/os_linux.cpp:4833]: It doesn't make sense to try and add an API that is exactly the same as `os::available_processors` except on Linux. This is a Linux and Linux container problem.

Comment (jerboaa) [reply src/hotspot/os/linux/os_linux.cpp:4833]: OK. Doing an alternative PR then.
